src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -110,6 +111,9 @@ public Task<RestoreResult> ExecuteAsync()
                 }
             }
 
+            // Validate, for noop this will be replayed from the assets file.
+            _success &= await ValidateProjectAsync(_request.Project, _logger);
```

Comment:
In the current code this can't be false coming into this check. If it were false this error should still be logged so that it provides consistent results. The check is also very cheap.